### DISCUSSION TITLE
Mobile backend development with Laravel

### DISCUSSION THREAD

**Discussion by devmsh** [laravel/framework#32758] [General]
Hi All,

I hope discussions will add a new home for Laravel developers to share their knowledge and ideas.

In the last years, we use Laravel for several use cases, but mostly to develop a mobile application backend API. and we end with a separated App repo in addition to Admin repo, where the first can be scaled up separately, and both projects connected to the same database.

As you know, Laravel comes with a powerful auth module of the box, but for mobile backend -especially with Mobile verification, devices info & FCM token, and SMS activation, it seems irrelevant and you need to redevelop your auth module yourself.

One of the most critical first-party packages in addition to Airlock/Sanctum for our use case is Laravel Telescope, I don't imagine myself debugging or monitoring my API without it, and we use it even in our production environment not only on local or starting server.

I'm really interested to know more about your experience or challenges when using Laravel as a Mobile App backend, what architecture you use? and what packages you usually install?

**Comment by glennjacobs** [comment]
I've started to develop API's in JSON:API. I much prefer this to regular RESTful API's and I find it much quicker due to this great package https://laravel-json-api.readthedocs.io/en/latest/

I feel it would be nice if Laravel could be installed ready to build an API, without all the Laravel Mix, Resources (e.g. JS+CSS), etc. I know it's not difficult to do yourself, but I feel more developers are building headless systems, and thinking API-First, so it might be nice to have.

**Comment by shadoWalker89** [comment]
I used laravel in a lot of my mobile apps.
For the json responses i used to use league fractal package before Resources were added to laravel, now i use the resources.
I use jwt for authentification https://github.com/tymondesigns/jwt-auth
The rest is fairly the same no matter what are you using laravel for.
That's the beauty of laravel you don't need to change much and everything works.

**Comment by emchooo** [comment]
@devmsh  Why do you separate App and Admin part for same project? How do you handle migrations? New models ?
